### PR TITLE
feature: [DEV-7047] Multi Visualization Component

### PR DESCRIPTION
**Pull request by joshlacey**
Storybook path: Pages/Dashboard/Multi Visualization

Testing steps from Editor:

Scenario: Tester has selected a dataset and have selected dashboard as the visualization type. They are now on the dashboard screen ready to apply the dataset to widgets.

1) drag and drop a table component on to the default widget row.
2) Configure the data for the row:
    option a) select the gear icon on the widget itself and select "apply to row" then continue the data configuration as normal.
    option b) select the gear icon on the row itself and configure the data.
3) before finishing the data configuration check the box that says: "Configure Multiple Visualizations"
4) when the dropdown appears select a category column you'd like to use to repeat your visualization.
5) complete the configuration.
6) navigate to the dashboard preview page.

Expected: You should see a table for each item in the column category that you selected.

### CODE REVIEW

**Comment by adamdoe** [comment]
@joshlacey I'm not able to select the data series or date/category columns when building a chart through the editor. Did you try building within the editor?

File: packages/editor/src/components/DataImport.tsx [hunk 1]
```diff
@@ -760,7 +760,7 @@ export default function DataImport() {
               </>
             )}
 
-            {showDataDesigner && <DataDesigner visualizationKey={null} dataKey={configureData.dataFileName} configureData={configureData} updateDescriptionProp={updateDescriptionProp} config={config} setConfig={setConfig} />}
+            {showDataDesigner && <DataDesigner visualizationKey={null} configureData={configureData} updateDescriptionProp={(key, value) => updateDescriptionProp(configureData.dataFileName, key, value)} config={config} setConfig={setConfig} />}
```
Comment (joshlacey) [inline packages/editor/src/components/DataImport.tsx:763]: @adamdoe this should fix the issue with the chart in editor.

**Comment by adamdoe** [comment]
@joshlacey bar charts are not receiving their color on dashboards when working through the editor.

**Comment by joshlacey** [comment]
> @joshlacey bar charts are not receiving their color on dashboards when working through the editor.

@Atash3000 do you have any ideas about why this would happen? I'm trying to test if it's related to the brush.data. I noticed when looking at the config that the data didn't seem to be mapped correctly in brush.data, but I can't figure out what's setting that in the first place.

**Comment by joshlacey** [comment]
> @joshlacey bar charts are not receiving their color on dashboards when working through the editor.

@adamdoe pushed up a commit that fixes this.

**Comment by adamdoe** [comment]
@joshlacey I'm still seeing errors when building out a dashboard. Here's one I'm seeing after choosing data and previewing the dashboard with the multivisualizations:

```
ErrorBoundary.jsx:16 TypeError: Cannot read properties of undefined (reading 'forEach')
    at CdcDashboardComponent.tsx:689:46
```

On option b, when configuring the data at the row level I end up with the same issue described above where I can't choose series or date/category data.

**Review by adamdoe — APPROVED** [review]

**Pull request merged**